datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1005,6 +1035,55 @@ pub async fn from_substrait_rex(
                 },
             })))
         }
+        Some(RexType::Subquery(subquery)) => match &subquery.as_ref().subquery_type {
+            Some(subquery_type) => match subquery_type {
+                SubqueryType::InPredicate(in_predicate) => {
+                    if in_predicate.needles.len() != 1 {
+                        Err(DataFusionError::Substrait(
+                            "InPredicate Subquery type must have exactly one Needle expression"
+                                .to_string(),
+                        ))
```

Comment:
I think you can use https://docs.rs/datafusion/latest/datafusion/common/macro.substrait_err.html to make the code less verbose

```suggestion
                       substrait_err!("InPredicate Subquery type must have exactly one Needle expression")
```